### PR TITLE
Alerting: Publish event when one or more rules are created or changed

### PR DESCRIPTION
**Pull request by rwwiv**
**What is this feature?**

This change publishes an event to the event bus when one or more rules are created or updated.

**Why do we need this feature?**

This allows external services to listen for this event as needed.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/store/alert_rule.go [hunk 1]
```diff
@@ -225,6 +225,7 @@ func (st DBstore) InsertAlertRules(ctx context.Context, rules []ngmodels.AlertRu
 			if _, err := sess.Insert(&ruleVersions); err != nil {
 				return fmt.Errorf("failed to create new rule versions: %w", err)
 			}
+			st.Bus.Publish(ctx, &RuleChangeEvent{})
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/store/alert_rule.go:228]: I wonder if we should log the error result.

**Comment by yuri-tceretian** [comment]
1. Can you please explain why we do not send events on delete? The name of the event implies that it should be sent on any change of the rules. If there is a good reason to not send on delete, then probably we should make the event name clearer.
2. What are your plans for the event listener side? In event-based approaches, it's usually a good idea to send the actual changes in the event. This will make the receiver much simpler and consistent because it will not need to fetch data from the store again.

**Comment by stevesg** [comment]
Agree having notifications of deletes, that would be valuable, and having the rule that changed in the event makes sense too.

**Review by yuri-tceretian — APPROVED** [review]
Linter does not like unchecked errors. 

Other than than LGTM

**Pull request merged**